src/Diagnostics.DataProviders/ConfigurationFactory.cs

```diff
@@ -174,18 +175,76 @@ private void LoadConfigurationValues(object dataProviderConfiguration)
                         continue;
                     }
 
-                    var value = GetValue(prefix, attribute.Name);
+                    dynamic value;
+                    string valueType;
+                  
+                    if (property.Name == "OverridableExceptions")
+                    {
+                        List<ITuple> overridableExceptionsList = new List<ITuple>();
+                        for (int i = 0; i < int.Parse(GetValue(prefix, "Retry:OverridableExceptionsToRetryAgainstLeaderCluster:AmountOfOverridableExceptions")); i++)
+                        {
+                            overridableExceptionsList.Add((
+                                GetValue(prefix, $"Retry:OverridableExceptionsToRetryAgainstLeaderCluster:Cases:{i.ToString()}:ExceptionString"),
+                                double.Parse(GetValue(prefix, $"Retry:OverridableExceptionsToRetryAgainstLeaderCluster:Cases:{i.ToString()}:MaxFailureResponseTimeInSeconds"))));
+                        }
 
-                    if (!string.IsNullOrWhiteSpace(value))
+                        valueType = "List<ITuple>";
+                        value = overridableExceptionsList;
+                    }
+                    else
                     {
-                        SetValue(dataProviderConfiguration, property, value, attribute.DefaultValue);
+                        valueType = "string";
+                        value = GetValue(prefix, attribute.Name);
+                    }
+                    
+                    if (valueType == "string")
+                    {
+                        if (!string.IsNullOrWhiteSpace(value))
+                        {
+                            SetValue(dataProviderConfiguration, property, value.ToString(), attribute.DefaultValue);
+                        }
+                    }
+                    else if(valueType == "List<ITuple>")
+                    {
+                        if (value != null)
+                        {
+                            SetValue(dataProviderConfiguration, property, value, attribute.DefaultValue);
+                        }
+                    }
+                    else
+                    {
+                        throw new Exception("InvalidType");
                     }
                 }
             }
         }
 
         protected abstract string GetValue(string prefix, string name);
 
+        protected void SetValue(object target, PropertyInfo property, object objectValue, object defaultValue)
+        {
+            object value = null;
+            if (property.PropertyType == typeof(string))
```

Comment:
Is it possible for the property.PropertyType to be something other than string/List<ITuple>?

Reply:
Before I started working on this change it was always string and anything else threw the error in the else block. I just added an option for List<ITuple> here because that was what I was using to get this piece to work.